app/routes/session.py

```diff
@@ -128,9 +151,76 @@ def login() -> Response:
 
     cookie_session["language_code"] = metadata.language_code
 
+    if (dataset_id := metadata["sds_dataset_id"]) and ru_ref:
```

Comment:
if the dataset id might not exist does this need to be a .get() ?

Reply:
We can use [] access because metadata here is using the MetadataProxy class